src/core/Akka/IO/Dns.cs

```diff
@@ -103,24 +110,28 @@ public class Resolved : Command
             public Resolved(string name, IEnumerable<IPAddress> ipv4, IEnumerable<IPAddress> ipv6)
             {
                 Name = name;
-                Ipv4 = ipv4;
-                Ipv6 = ipv6;
+                Ipv4 = ipv4?.ToImmutableList() ?? ImmutableList<IPAddress>.Empty;
+                Ipv6 = ipv6?.ToImmutableList() ?? ImmutableList<IPAddress>.Empty;
 
-                _addr = ipv4.FirstOrDefault() ?? ipv6.FirstOrDefault();
+                _addr = Ipv4.FirstOrDefault() ?? Ipv6.FirstOrDefault();
             }
 
+            public bool IsSuccess => Exception == null;
+            
+            public Exception Exception { get; }
```

Comment:
Need to add this new property to pass back exceptions from failed DNS resolve call